cmake/detect-arch.c

```diff
@@ -101,10 +101,6 @@
         #error archfound riscv32
     #endif
 
-// Emscripten (WebAssembly)
```

Comment:
Is removing this necessary? I wonder if it is used by configure script.

Reply:
Removing is not necessary, but I figured this part is also unnecessary now that CMake has its own detection.

Good point about configure script though, I haven't looked into how that's set up - is it not generated from CMake?

Reply:
> is it not generated from CMake?

I see it's not. OTOH it doesn't seem to use detect-arch.c either, it just asks user to provide `ARCH=...` manually if not auto-detected.

I suppose that's why this C file is in the `cmake` folder, since CMake config is the only one that uses it.

Reply:
I agree, however I think we should leave these lines in there in case things change in the future.

Reply:
I reverted this part.